node/full_node_integration_test.go

```diff
@@ -189,6 +189,96 @@ func TestLazyAggregator(t *testing.T) {
 	require.NoError(waitForAtLeastNBlocks(node, 4, Header))
 }
 
+// TestFastDASync verifies that nodes can sync DA blocks faster than the DA block time
+func TestFastDASync(t *testing.T) {
+	// Test setup, create require and contexts for aggregator and client nodes
+	require := require.New(t)
+	aggCtx, aggCancel := context.WithCancel(context.Background())
+	defer aggCancel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Set test params
+	clientNodes := 2
+	bmConfig := getBMConfig()
+	// Set the DABlockTime to a large value to avoid test failures due to
+	// slow CI machines
+	bmConfig.DABlockTime = 1 * time.Second
+	// Set BlockTime to 2x DABlockTime to ensure that the aggregator node is
+	// producing DA blocks faster than rollup blocks. This is to force the
+	// block syncing to align with hard confirmations.
+	bmConfig.BlockTime = 2 * bmConfig.DABlockTime
+	const numberOfBlocksToSyncTill = 5
+
+	// Create the 2 nodes
+	nodes, _ := createNodes(aggCtx, ctx, clientNodes, bmConfig, t)
+	node1 := nodes[0]
+	node2 := nodes[1]
+
+	// Start node 1
+	require.NoError(node1.Start())
+	defer func() {
+		require.NoError(node1.Stop())
+	}()
+
+	// Wait for node 1 to sync the first numberOfBLocksToSyncTill
+	require.NoError(waitForAtLeastNBlocks(node1, numberOfBlocksToSyncTill, Store))
+
+	// Now that node 1 has already synced, start the second node
+	require.NoError(node2.Start())
+	defer func() {
+		require.NoError(node2.Stop())
+	}()
+
+	// Start and launch the timer in a go routine to ensure that the test
+	// fails if the nodes do not sync before the timer expires
+	ch := make(chan struct{})
+	defer safeClose(ch)
+	// Set the timer to expire 1 block before the numberOfBlocksToSyncTill
+	timer := time.NewTimer((numberOfBlocksToSyncTill - 1) * bmConfig.DABlockTime)
```

Comment:
We can probably reduce this to 2 blocks.

Reply:
Reduced it to a little over 1 block to be stricter.